### PR TITLE
Setting user agent on REST API requests

### PR DESCRIPTION
**Pull request by bhariharan**
Previously we were relying on `CSFAction` to set the `User-Agent` header. Now we have to set it at the `RestAPI` layer.

### CODE REVIEW

**Comment by wmathurin** [comment]
So we have been tracking rest api calls all of 5.1?

**Comment by huminzhi** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Code coverage under minimum of 80%</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
SFLoggerTests | testExtraLoggers|
SFLoggerTests | testExtraLoggers|
SFLoggerTests | testLogFormatter|


## Current coverage for SalesforceSDKCore.framework is `48%`
Files changed | - | - 
--- | --- | ---
SFRestRequest.m | `77%` | :warning:

---
## Current coverage for SalesforceAnalytics.framework is `42%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by bhariharan** [comment]
Yeah, looks like it :-(

**Comment by wmathurin** [comment]
Should we put that in 5.2 (maybe just move the tag)?

**Comment by bhariharan** [comment]
Yeah - I can do that.

**Comment by bhariharan** [comment]
I can't add the tests right now, since we don't expose the request headers. I'll add the tests and expose the headers in `6.0`, but just add the fix without the tests in `5.2` (to minimize changes).

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.m [hunk 1]
```diff
@@ -164,6 +165,11 @@ - (NSURLRequest *)prepareRequestForSend {
             [self.request setValue:bearer forHTTPHeaderField:@"Authorization"];
         }
 
+        // Sets Mobile SDK user agent on REST API requests, if it hasn't been set already by .
+        if (self.request.allHTTPHeaderFields && self.request.allHTTPHeaderFields.allKeys && ![self.request.allHTTPHeaderFields.allKeys containsObject:@"User-Agent"]) {
```
Comment (khawkins) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.m]: Not to necessarily change the code, but the following test:
```obj-c
if (![self.request.allHTTPHeaderFields.allKeys containsObject:@"User-Agent"]) {
}
```
is all that's necessary in Objective-C.  Any of the upstream properties being `nil` will automatically evaluate to 'false' from that evaluation.

Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestRequest.m]: Sounds good!

**Comment by bhariharan** [comment]
@khawkins Fixed [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/2127/commits/24968a1c0eb8a97f7df0a097bb90c8cad2994b39).

**Pull request merged**